### DISCUSSION TITLE
Optional events

### DISCUSSION THREAD

**Discussion by tachigami** [spatie/laravel-settings#104] [Ideas]
I use the LaravelSettings as a dynamic alternative to a laravel's static config files to change app behavior in runtime. Is there any chances to make events optional in future releases? I find that it fires way too much "SettingsLoaded" events on every request. Especially when the Settings are called in a ServiceProvider. It even starts to fire events for Http OPTIONS requests. I don't see any reasons to have these events unless needed. It could be an option in settings.php to switch them on/off.

**Comment by rubenvanassche** [comment]
I don't see this as a problem. If I'm not mistaken, Eloquent fires like a 100 events for each request I make. As long as you don't listen to this events there shouldn't be a big performance impact. But correct me if I'm wrong.

**Reply by tachigami** [reply]
Haha. I actually thought about it and I couldn't agree more. Seems there's nothing to worry about for now.
